app/uk/gov/hmrc/thirdpartyapplication/services/ApplicationUpdateService.scala

```diff
@@ -49,4 +54,21 @@ class ApplicationUpdateService @Inject()(
       case _ => Future.successful(Validated.invalidNec(s"Unknown ApplicationUpdate type $applicationUpdate"))
     }
   }
+
+  private def sendAdviceEmail(events: NonEmptyList[UpdateApplicationEvent])(implicit hc: HeaderCarrier): Future[HasSucceeded] = {
+    def sendEmail(event: UpdateApplicationEvent) = {
+      if (event.emailAdvice) {
+        event match {
+          case evt: UpdateApplicationEvent.NameChanged => nameChangeEventHdlr.sendAdviceEmail(evt)
+          case _ => throw new RuntimeException(s"UnexpectedEvent type for emailAdvice ${event}")
+        }
+      } else {
+        Future.successful(HasSucceeded)
+      }
+    }
+    events match {
+      case NonEmptyList(e, Nil) => sendEmail(e)
+      case NonEmptyList(e, tail) => sendEmail(e).flatMap(_ => sendAdviceEmail(NonEmptyList.fromListUnsafe(tail)))
```

Comment:
If there are multiple events and the first one fails it looks like we don't attempt to process subsequent ones, is that the behaviour we want?